src/MongoDB/Manager.c

```diff
@@ -538,6 +538,42 @@ static PHP_METHOD(Manager, selectServer)
 	}
 } /* }}} */
 
+/* {{{ proto MongoDB\Driver\Session MongoDB\Driver\Manager::startSession([array $options = null])
+   Returns a new client session */
+static PHP_METHOD(Manager, startSession)
+{
+	php_phongo_manager_t    *intern;
+	zval                    *options = NULL;
+	mongoc_session_opt_t    *cs_opts = NULL;
+	mongoc_client_session_t *cs;
+	bson_error_t             error;
+	SUPPRESS_UNUSED_WARNING(return_value_ptr) SUPPRESS_UNUSED_WARNING(return_value_used)
+
+
+	intern = Z_MANAGER_OBJ_P(getThis());
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "|a!", &options) == FAILURE) {
+		return;
+	}
+
+	if (options && php_array_exists(options, "causalConsistency")) {
```

Comment:
Do you want to extract option parsing into its own function like we do for many other options arrays?

Reply:
We can consider that in the future if the options list grows a bit. I believe I modeled this after the ReadPreference constructor, which also parses a single option ("maxStalenessSeconds") at present.